src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -304,37 +307,52 @@ private static Stream ApplyFilter(string csdl, string entitySetOrSingleton, XslC
             }
         }
 
+        public static IConfiguration GetConfiguration(string settingsFile)
+        {
+            settingsFile ??= "appsettings.json";
+
+            IConfiguration config = new ConfigurationBuilder()
+            .AddJsonFile(settingsFile, true)
+            .Build();
+
+            return config;
+        }
+
         /// <summary>
         /// Converts CSDL to OpenAPI
         /// </summary>
         /// <param name="csdl">The CSDL stream.</param>
         /// <returns>An OpenAPI document.</returns>
-        public static async Task<OpenApiDocument> ConvertCsdlToOpenApi(Stream csdl)
+        public static async Task<OpenApiDocument> ConvertCsdlToOpenApi(Stream csdl, string settingsFile = null)
         {
             using var reader = new StreamReader(csdl);
             var csdlText = await reader.ReadToEndAsync();
             var edmModel = CsdlReader.Parse(XElement.Parse(csdlText).CreateReader());
+            
+            var config = GetConfiguration(settingsFile);
+            var settings = config.GetSection("OpenApiConvertSettings").Get<OpenApiConvertSettings>();
```

Comment:
why not just binding to avoid multiple defaults? (i.e. the defaults from the lib, the defaults in the configuration file, the defaults below)

```CSharp

var settings = new OpenApiConvertSettings{ // all the default values)
config.GetSecion("OpenApiConvertSettings).Bind(settings);
```

Reply:
Does binding the default settings allow one to customize them by providing their own settings file?